### PR TITLE
[doc] added clarification for Processors about TagObjects

### PR DESCRIPTION
**Pull request by renatosc**
Fixes #
[Design discussion issue #](https://github.com/open-telemetry/opentelemetry-dotnet/discussions/5395)

## Changes

Added line to the doc clarifying that Processors needs to use TagObjects instead of Tags to access all tags (in the same style as seen in the Exporters section)

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by vishweshbankwar — APPROVED** [review]
Thanks @renatosc!

**Review by cijothomas — APPROVED** [review]

File: docs/trace/extending-the-sdk/README.md [hunk 1]
```diff
@@ -256,6 +256,8 @@ Custom processors can be implemented to cover more scenarios:
   and `OnShutdown`.
 * `OnStart` and `OnEnd` should be thread safe, and should not block or take long
   time, since they will be called on critical code path.
+* Processors should use `Activity.TagObjects` collection instead of
```
Comment (reyang) [inline docs/trace/extending-the-sdk/README.md:259]: Minor - maybe link to the .NET official doc?

Comment (renatosc) [reply docs/trace/extending-the-sdk/README.md:259]: That is a good idea. I will update the PR.

Comment (utpilla) [reply docs/trace/extending-the-sdk/README.md:259]: I think, as a best practice, we should encourage users to use `Activity.EnumerateTagObjects` instead. That's more efficient than simply enumerating over `Activity.TagObjects`.

Comment (renatosc) [reply docs/trace/extending-the-sdk/README.md:262]: Updated

**Review by cijothomas — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

**Pull request merged**